tools/NetCorePsd1Sync/NetCorePsd1Sync/Utility/StringExtensions.cs

```diff
@@ -0,0 +1,276 @@
+﻿// ----------------------------------------------------------------------------------
+//
+// Copyright Microsoft Corporation
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+// http://www.apache.org/licenses/LICENSE-2.0
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+// ----------------------------------------------------------------------------------
+
+using System;
+using System.Collections.Generic;
+using System.ComponentModel;
+using System.Linq;
+using System.Linq.Expressions;
+using NetCorePsd1Sync.Model;
+using static NetCorePsd1Sync.Model.PsDefinitionConstants;
+using static NetCorePsd1Sync.Utility.AttributeHelper;
+
+namespace NetCorePsd1Sync.Utility
+{
+    internal static class StringExtensions
+    {
+        private static bool GetDescription<TClass, TProp>(Expression<Func<TClass, TProp>> propertySelector, out string description)
+        {
+            var hasDescription = TryGetPropertyAttributeValue<TClass, TProp, DescriptionAttribute, string>(propertySelector, attr => attr.Description, out description);
+            description = hasDescription ? $"{CommentPrefix}{description}" : null;
+            return hasDescription;
+        }
+
+        private static (TProp value, string firstLeader) GetValue<TClass, TProp>(TClass @class, Expression<Func<TClass, TProp>> propertySelector)
```

Comment:
Let's not use '@class' as the type variable here, lets use TargetClass, or use another descriptive Prefix rather than the reserved word.

Reply:
Fixed.